test/conftest.py

```diff
@@ -354,6 +354,23 @@ def embed_documents(self, documents):
         return np.random.rand(len(documents), self.embedding_dim)
 
 
+class MockFilterRetriever(MockRetriever, FilterRetriever):
+    def __init__(self, document_store: BaseDocumentStore, mock_document: Document):
+        self.document_store = document_store
+        self.mock_document = mock_document
+
+    def retrieve(
+        self,
+        query: str,
+        filters: dict,
+        top_k: Optional[int],
+        index: str,
+        headers: Optional[Dict[str, str]],
+        scale_score: bool,
+    ):
+        return [self.mock_document]
```

Comment:
This one can be moved in `test_retriever.py`, so it's close to the test where it's used.

Reply:
Removed old test and added mock to the test file, right before the test itself